Sources/Gir2SwiftPlugin/plugin.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import PackagePlugin
+
+extension Path {
+    func exists() -> Bool {
+        FileManager.default.fileExists(atPath: string)
+    }
+}
+
+let module = targetBuildContext.moduleName
+let outputDir = targetBuildContext.outputDirectory.appending("Generated")
+
+let output = outputDir.appending("Output.swift")
+
+func packageDirectory(for target: Path) -> Path {
+    var curr = target
+    while !curr.appending("Package.swift").exists() {
+        if curr.stem.isEmpty {
+            fatalError("Could not find Package.swift for target \(target)")
+        }
+        curr = curr.removingLastComponent()
+    }
+    return curr
+}
+
+let girNames = try targetBuildContext.dependencies.compactMap { target -> String? in
+    let package = packageDirectory(for: target.targetDirectory)
+    let manifest = package.appending("gir2swift-manifest.sh")
+    guard manifest.exists() else { return nil }
+
+    let pipe = Pipe()
+
+    let proc = Process()
+    proc.executableURL = URL(fileURLWithPath: manifest.string)
+    proc.arguments = ["gir-name"]
+    proc.standardOutput = pipe
+    try proc.run()
+
+    return String(data: pipe.fileHandleForReading.readDataToEndOfFile(), encoding: .utf8)?
+        .trimmingCharacters(in: .whitespacesAndNewlines)
+}
+
+let girSearchPaths: [Path] = ["/opt/homebrew/share/gir-1.0", "/usr/local/share/gir-1.0", "/usr/share/gir-1.0"]
+guard let girPath = girSearchPaths.first(where: { searchPath in
+    girNames.allSatisfy { gir in searchPath.appending("\(gir).gir").exists() }
+}) else {
+    fatalError("Could not locate GIR path")
+}
```

Comment:
Have you tested this on M1 macs? We have an issue on mac silicon that was not fixed in the mainstream `gir2swift generation driver yet` https://github.com/rhx/SwiftGtk/issues/45. **Summary:** On the M1 macs at the time, we learned, that all gir files are **not** being saved into `"/opt/homebrew/share/gir-1.0"`. Instead, they are all stored at the lib folder. 

I have already created a fix, but I haven't merget it yet, and forgot about it (to be honest :) )  You can see the fixed function here https://github.com/mikolasstuchlik/gir2swift/blob/9d5ddc18d7773962e0336ab09b6588fdedc9731a/gir2swift-generation-driver.sh#L27

Reply:
Interesting, this feels non-trivial so IMO it's another reason we should move the determination of the GIR directory into gir2swift or into something else which is executed at build time rather than when the plugin is run. From my understanding, plugins themselves need to be as simple as possible (plus moving it to build time would hopefully solve any sandboxing issues).